pyccel/codegen/pipeline.py

```diff
@@ -325,20 +330,34 @@ def handle_error(stage):
                     # remove library folder to avoid missing files and copy
                     # new one from pyccel stdlib
                     lib_dest_path = os.path.join(pyccel_dirpath, lib_name)
-                    if os.path.exists(lib_dest_path):
-                        shutil.rmtree(lib_dest_path)
-                    shutil.copytree(lib_path, lib_dest_path)
+                    if not os.path.exists(lib_dest_path):
+                        try:
+                            shutil.copytree(lib_path, lib_dest_path)
+                        except FileExistsError:
+                            pass
```

Comment:
Are you trying to copy only the minimum number of files here? If yes:

- do you see a significant speedup in doing so?
- the two lines of comment just above should be updated.

Reply:
The reason for avoiding the copy is not to reduce the number of files that are copied around but is to avoid one thread modifying a file (by overwriting it) while another thread is using it (compiling). I'm not sure if the copy itself is a problem, but previously we were deleting before copying and this was definitely creating a race condition

Reply:
Fixed in c012c7fc

Reply:
I see, thanks